src/apps/intermediate-list/stackable-fees/fee-details-content.tsx

```diff
@@ -0,0 +1,130 @@
+  const { termsOfTradeUrl } = useUrls();
+  const handleAcceptedTerms = () => {
+    setCheck(!check);
+  };
+  const {
+    amount = 0,
+    creationDate = "",
+    dueDate = "",
+    materials = []
+  } = feeDetailsData;
+  const prePaymentTypeChange = isDateBeforePaymentChangeDate(new Date(dueDate));
+  const creationDateFormatted = dayjs(creationDate).format("D. MMMM YYYY");
+
+  return (
+    <div className="modal modal-show modal-loan">
+      <div className="modal__screen-reader-description" id="describemodal">
+        {t("feeDetailsModalScreenReaderText")}
+      </div>
+      <div className="modal-loan__container">
+        <div className="modal-loan__header">
+          <div className="mr-32">
+            <FeeStatusCircle
+              dueDate={dueDate || ""}
+              feeCreationDate={creationDate}
+            />
+          </div>
+          <div>
+            <h2 className="modal-loan__title text-header-h2">
+              {t("turnedInText", {
+                placeholders: { "@date": creationDateFormatted }
+              })}
+            </h2>
+          </div>
+        </div>
+        <div className="modal-loan__buttons">
+          <CheckBox
+            id="checkbox_id__fee_details"
+            onChecked={() => handleAcceptedTerms()}
+            label={
+              <>
+                {t("iAcceptText")}{" "}
+                <Link href={new URL(termsOfTradeUrl)}>
+                  {t("termsOfTradeText")}
+                  <sup>*</sup>
+                </Link>
+              </>
+            }
+          />
+          <div>
+            <p>{amount},-</p>
+          </div>
+          {prePaymentTypeChange && !check && (
+            <button
+              type="button"
+              className="btn-primary btn-outline btn-small arrow__hover--right-small"
+              disabled
+            >
+              {t("payText")}
+            </button>
+          )}
+          {prePaymentTypeChange && check && (
+            <button
+              type="button"
+              className="btn-primary btn-outline btn-small arrow__hover--right-small"
+              disabled
+            >
+              {t("payText")}
+            </button>
+          )}
+          {!prePaymentTypeChange && !check && (
+            <button
+              type="button"
+              className="btn-primary btn-outline btn-small arrow__hover--right-small"
+              disabled
+            >
+              {t("payText")}
+            </button>
+          )}
+          {!prePaymentTypeChange && check && (
+            <Link
+              className="btn-primary btn-filled btn-small arrow__hover--right-small"
+              href={
+                new URL(
+                  "https://www.borger.dk/vaelg-kommune?actionPageId=065ca8f9-a1f5-4946-ada7-12e163f568df&selfserviceId=7200a519-38ad-48b9-b19a-6e5783e39999"
+                )
+              }
+              isNewTab
+            >
+              {t("payText")}
```

Comment:
I wrote another comment about a similar situation in another component. Please look at it. I think you are repeating to much code. 
And the url should definitely not be hardcoded. If it is temporary at least a TODO is required.

Reply:
I reduced the footprint of this, also using the core concept of your earlier suggestion.